thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/models/DimensionTreeMapModel.js

```diff
@@ -26,6 +26,11 @@ DimensionTreeMapModel.prototype = {
         this.metricId = params.metric.id;
         this.metricName = params.metric.name;
       }
+
+      if (params.metricId) {
```

Comment:
I think we can run into the case where both params.metricId and params.metric are undefined. so `params.metric.id` will throw a TypeError.

Ideally, I'd like to only keep `params.metricId` and deprecate `params.metric`. This will require a deeper dive into the code. I'll keep this section of code as-is, and will update the 'to-dos'